libraries/Microsoft.Bot.StreamingExtensions/IContentStream.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Implemented by stream attachments compatible with the Bot Framework Protocol 3 with Streaming Extensions.
+    /// </summary>
+    public interface IContentStream
+    {
+        /// <summary>
+        /// Gets a guid to use as the unique identifier of this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Guid.
+        /// </value>
+        Guid Id { get; }
+
+        /// <summary>
+        /// Gets or sets the name of the type of the object contained within this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Plain text type name.
+        /// </value>
+        string Type { get; set; }
+
+        /// <summary>
+        /// Gets or sets the length of this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Null or a numeric value bound by Min/Max Int.
+        /// </value>
+        int? Length { get; set; }
+
+        /// <summary>
+        /// Method called to retrieve the data contained within this ContentStream.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="Stream"/> of data.
+        /// </returns>
+        Stream GetStream();
+    }
```

Comment:
Why is the return type Stream? WE're in a "ContentStream" interface, shouldn't this be ContentStream?

Reply:
Changed to a property and refactored. A content stream is a stream with the necessary properties to be processed as content. Changing the class name would probably make more sense, most of our 'streams' get their name from wrapping data sent as members of the 'Streams' collection on requests. The type of stream, 'Content', 'Payload', etc, is really describing the additional information provided along with the real stream, and explains how it should be handled.